backend/index.py

```diff
@@ -79,7 +81,40 @@ def clean_expired_tokens():
 app.register_blueprint(tag_bp, url_prefix='/tag')
 
 
+socketio = SocketIO(app, cors_allowed_origins="*")
+
+users = {}
+
+@socketio.on('join_room')
+def handle_join_room(data):
+    authorized_username = data.get("authorizedUsername")
+    users[authorized_username] = request.sid
+
+@socketio.on('send_message')
+def handle_send_message(data):
+    targetUsername  = data.get("targetUsername")
+    authorizedUsername = data.get("authorizedUsername")
+    recipient_session_id = users.get(targetUsername)
+    message = data.get('message')
+    
+    if recipient_session_id:
+        emit("new_private_message", {"sender": authorizedUsername, "message": message, "recipient": targetUsername}, room=recipient_session_id)
+
+@socketio.on('connect')
+def handle_connect():
+    print('Client connected')
```

Comment:
In my opinion, I don't think you need these prints (this and line 113). If you want to print them, you can consider using logging. Otherwise LGTM.